docs/DeveloperGuide.md

```diff
@@ -4,26 +4,231 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Setting Up & Getting Started
+
+1. Fork the repo at https://github.com/AY2324S1-CS2113T-W11-1/tp.
+2. Clone the fork into your computer.
+
+## Design & Implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+<img src="images/ArchitectureDiagram.png" />
+
+
+**Main components of SysLib Architecture**
+
+SysLib currently consists of four main components:
+
+- `UI`: User Interaction
+- `Parser`: Parsing User Response 
+- `Command`: Command Executor
+- `Data`: Holds the data of SysLib
+
+
+   
+### UI Component
+
+### Parser Component
+
+### Command Component
+
+
+### Data Component
+
+## Implementation 
+This section provides details on how certain features are implemented. 
+
+### Find Resource Feature
+
+The `find` command allows users to search for resources based on specified filters such as author (`/a`), ISBN (`/i`), ID (`/id`), and title (`/t`). The results will show all resources that match any of the given filters.
+
+`find` has the following options:
+- `find /id [ID]`
+- `find /t [TITLE]`
+- `find /a [AUTHOR]`
+- `find /i [ISBN]`
```

Comment:
According to the [textbook](https://se-education.org/guides/conventions/markdown.html), we are suggested to use `*` instead of `-` for bullet-points, but it is optional to follow it.